edb/pgsql/ast.py

```diff
@@ -105,7 +105,8 @@ def _infer_nullability(self, kwargs: typing.Dict[str, object]) -> bool:
         for v in kwargs.values():
             if typeutils.is_container(v):
                 items = typing.cast(typing.Iterable, v)
-                nullable = any(getattr(vv, 'nullable', False) for vv in items)
+                # FIXME: Is this right???
+                nullable = all(getattr(vv, 'nullable', False) for vv in items)
```

Comment:
This is not true.

Counter example:
```
POWER(4, NULL)
```

`4` is not nullable, so `all(False, True)` will be `False`, but `POWER(4, NULL)` is nullable.

This should be a proper AST pass, not this kwargs hackey.

Reply:
`any` in most cases and `all` in the case of `COALESCE`